onnxruntime/test/framework/execution_frame_test.cc

```diff
@@ -259,128 +259,6 @@ TEST_F(ExecutionFrameTest, MemPatternTest) {
   ASSERT_EQ(p->GetBlock(4)->offset_, kAllocAlignment);
 }
 
-#ifdef ENABLE_TRAINING
```

Comment:
@SherlockNomad Do we need to replace this with a different test to cover the "external outputs" path (now that YieldOp is a no-op)?

Reply:
yep... YieldOp is the only op in master that sets .ExternalOutput()

Maybe check PythonOp?

Reply:
PythonOp isn't in master yet, its under development, it might make sense for PythonOp PR to add a test for external outputs.

Reply:
I will leave a part of this test since we are keeping SetExternalOutput API but will remove YieldOp/fw/bw pass related tests since those code paths no longer use this API.